testing/pr-testing-guidelines-and-checklists.md

```diff
@@ -0,0 +1,149 @@
+# PR testing guidelines and checklists 
+
+## Introduction
+
+This proposal should cover the PR testing of the repositories concerning the new Aerogear/Feedhenry developments.
+It was based on a discussion between Adam Saleh, Vitali Chepeliuk, Pavel Sturc and Vojtech Sazel.
+
+## Problem Description
+
+We currently don't have a proper set of guidelines for our PR checks and previously we used them mostly just to produce the build-artifacts
+in some fashion. The quality checks have often been added in ad-hoc manner and then disregarded/forgotten about.
+
+## Expectations
+
+If we create a set of coherent guidelines and model our automated PR checks by them we expect
+- to catch regressions early
+- to maintain a good quality of incoming code
+
+## Terms
+
+* relevant repository - repository that could influence ship-able artifacts by it's contents in a bad way. If we really don't want to follow these guidelines, just claim that the repository is currently irrelevant for our project.
+  
+## Proposed Solution
+
+These proposals only concern relevant repositories.
+
+### Repository should have a mandatory review by other person
+
+This is easily enabled in GitHub. It mandates that at least one person gives a PR a green check-mark.
+
+There will be some decisions that we don't want to automate on purpose, for example, we will probably display code-coverage prominently,
+but we won't have any automatically enforced that would stop PR from getting merged due to low coverage.
+
+In the future, we might have optional integration tests triggered by comments or specific tags on the PR. This would be hard to enforce automatically,
+so mandating those would fall as a responsibility of the reviewer.
+
+There could be a template with a checklist delivered by a .github/PULL_REQUEST_TEMPLATE.md
+Checklist would include:
+
+* Read through the code and ask about any unclear code
+* Try to checkout and run the relevant code
```

Comment:
If we have good enough CI we shouldn't need this

Reply:
True, will reword this to specifically mandate we have good enough CI that reviewers don't need to run the code :-)